### PR TITLE
[corlib] Change the RuntimeModule.GetGuid and RuntimeAssembly.GetAotId icalls to return a byte array instead of a string, so it doesn't have to be parsed by the Guid ctor. Also allocate the arrays in the c# code and pass them to the icalls.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/metadata/icall.c [hunk 1]
```diff
@@ -5066,15 +5066,20 @@ ves_icall_System_Reflection_RuntimeAssembly_GetManifestResourceNames (MonoReflec
...
-	return res;
+	g_assert (mono_array_handle_length (guid_h) == 16);
+
+	guint8 *data = (guint8*) mono_array_addr_with_size_internal (MONO_HANDLE_RAW (guid_h), 1, 0);
+
+	guint8 *aotid = mono_runtime_get_aotid_arr ();
+	if (!aotid) {
+		return FALSE;
+	} else {
+		memcpy (data, aotid, 16);
```
Comment (lambdageek) [inline mono/metadata/icall.c]: 1. Better to move `data` closer to where it's used.
2. Wrap the place where `data` is used in `MONO_ENTER_NO_SAFEPOINTS`/`MONO_EXIT_NO_SAFEPOINTS`

```
    MONO_ENTER_NO_SAFEPOINTS;
    guint8 *data = (guint8*) mono_array_addr_with_size_internal (MONO_HANDLE_RAW (guid_h), 1, 0);
    memcpy (data, aotid, 16);
    MONO_EXIT_NO_SAFEPOINTS;
```

File: mono/metadata/icall.c [hunk 2]
```diff
@@ -6285,12 +6290,21 @@ ves_icall_System_Reflection_RuntimeModule_GetGlobalType (MonoImage *image, MonoE
...
+	g_assert (mono_array_handle_length (guid_h) == 16);
 
-	return mono_string_new_handle (domain, image->guid, error);
+	guint8 *data = (guint8*) mono_array_addr_with_size_internal (MONO_HANDLE_RAW (guid_h), 1, 0);
+
+	if (!image->metadata_only) {
+		g_assert (image->heap_guid.data);
+		g_assert (image->heap_guid.size >= 16);
+
+		memcpy (data, (guint8*)image->heap_guid.data, 16);
```
Comment (lambdageek) [inline mono/metadata/icall.c]: Same as other function - move decl of `data` down and wrap the use in `NO_SAFEPOINTS`

**Comment by jaykrell** [comment]
Can’t you use fixed and pass byte*?

**Comment by jaykrell** [comment]
(And length)

**Comment by vargaz** [comment]
@monojenkins build deb with monolite

**Comment by jaykrell** [comment]
I can do the fixed/*/length change after. :)

**Comment by vargaz** [comment]
@monojenkins build failed

**Pull request merged**